src/measure_theory/integration.lean

```diff
@@ -128,6 +128,14 @@ is_measurable_cut (λ _ b, b ∈ s) f (λ b, is_measurable.const (b ∈ s))
 protected theorem measurable [measurable_space β] (f : α →ₛ β) : measurable f :=
 λ s _, is_measurable_preimage f s
 
+protected lemma sum_measure_preimage_singleton (f : α →ₛ β) {μ : measure α} (s : finset β) :
+  ∑ y in s, μ (f ⁻¹' {y}) = μ (f ⁻¹' ↑s) :=
```

Comment:
This can be generalized to `f` being an arbitrary measurable map (and moved to `measure_space`), right?

Reply:
There is a version in `measure_space`. It assumes `∀ y ∈ s, is_measurable (f ⁻¹' {y})`. I can't prove it for any `measurable f` unless I also assume `measurable_singleton_class β`.

Reply:
Ah, you're right.